test/net/sourceforge/kolmafia/request/UseItemRequestTest.java

```diff
@@ -302,6 +306,56 @@ void maxUsesWorksForHtmlSpleenItemsWithConsumptionType(int spleenUsed, int maxUs
     }
   }
 
+  @Nested
+  class zeroFullness {
+    @ParameterizedTest
+    @CsvSource({"5140", "10883"})
+    void itShouldNotDivideByZeroWhenConsumingAstralEnergyDrink(int itemID) {
+      int spleenUsed = 0;
+      int maxUses = Integer.MAX_VALUE;
+      var cleanups =
+          new Cleanups(
+              withSpleenUse(spleenUsed),
+              withHP(0, 10, 10),
+              withMP(0, 10, 10),
+              withPath(AscensionPath.Path.STANDARD));
+
+      try (cleanups) {
+        assertThat(UseItemRequest.maximumUses(itemID, ConsumptionType.SPLEEN), is(maxUses));
```

Comment:
it shouldn't be max uses for the astral energy drink, though? If it's the 8 one it should be 1, if it's the 5 one it should be 3.

Reply:
The test as written always looks for the raw spleen hit of "astral energy drink".  That is null and thus zero is used upstream.  Since the test was designed to show a div by zero no longer occurs, it shows that the error has been fixed.

That said there is a different test that would meet your expectations.  I'll look at writing it.

Reply:
I'm having a problem with that test.

UseItemRequest has
`  public static final int maximumUses(String itemName) {
    int itemId = ItemDatabase.getItemId(itemName);
    if (itemId > 0) {
      itemName = ItemDatabase.getItemDataName(itemId);
    }
    return UseItemRequest.maximumUses(itemId, itemName, ConsumptionType.NONE, false);`

Using the debugger, if it is passed "[10883]astral energy drink" it correctly determines the itemId of 10883. ItemDatabase.getItemDataName(itemId) then returns "astral energy drink".  The end result is zero maximum uses because "astral energy drink" is not the name of a known, existing consumable, and you can't consume what doesn't exist.